app/models/group.rb

```diff
@@ -10,25 +10,65 @@
 class Group < ApplicationRecord
   belongs_to :leader, class_name: 'User'
   belongs_to :draw
-  has_many :members, class_name: 'User'
+  has_many :memberships
+  has_many :members, through: :memberships, source: :user
+
+  enum status: %w(open full locked)
 
-  validates :size, presence: true, numericality: { only_integer: true,
-                                                   greater_than: 0 }
-  validates :status, presence: true
-  validates :leader, presence: true
   validates :draw, presence: true
+  validates :status, presence: true
+  validates :size, presence: true,
+                   inclusion: { in: ->(g) { g.draw.suite_sizes },
+                                if: ->(g) { g.draw.present? } }
+  validates :leader, presence: true, inclusion: { in: ->(g) { g.members } }
 
-  enum status: %w(open full locked)
+  validate :validate_members_count, if: ->(g) { g.size.present? }
+  validate :validate_status, if: ->(g) { g.size.present? }
 
-  after_save :add_leader_to_members
+  before_validation :add_leader_to_members, if: ->(g) { g.leader.present? }
 
   def name
     "#{leader.name}'s Group"
   end
 
+  # Updates the status to match the group size (open when fewer members than
+  # the size, and full when they are the same)
+  def update_status
+    if memberships_count < size
+      update_attributes(status: 'open')
```

Comment:
While working on #51 I discovered that `update` is an alias for `update_attributes`, so since it's both shorter and cleaner why don't we use that and continue moving forward?